### DISCUSSION TITLE
plotGSEA Term descriptions

### DISCUSSION THREAD

**Discussion by Natpod** [GangLiLab/genekitr#16] [Ideas]
Hi, 

Thank you for all the work you have put in in genekitr, this is a wonderful tool that give very insightful representations from my data.

I thought about a new functionality from the plotting module of the package. I think the plot enrichment function plotGSEA can benefit from plotting the descriptions or full names of the gene categories. This can be provided in a separate argument for example.

**Comment by reedliu** [comment]
Hi, glad you like it.
- For `plotEnrich()` function, users can directly modify the **'Description'** column in the results of ORA. 
For example,
```
data(geneList, package = "genekitr")
id <- names(geneList)[abs(geneList) > 1.5]
logfc <- geneList[id]
gs <- geneset::getGO(org = "human",ont = "bp")
ego <- genORA(id, geneset = gs)
ego <- ego[1:10, ]

# default plot
plotEnrich(ego, plot_type = "dot")
```
<img width="499" alt="image" src="https://github.com/GangLiLab/genekitr/assets/34302878/7179a1a8-605a-4565-96a4-56c98722b9e7">

```
# if user wants to combine GO id and term name, just change the "Description" column
ego$Description = paste0(ego$Hs_BP_ID,'(',ego$Description,')')
plotEnrich(ego, plot_type = "dot")

```
<img width="495" alt="image" src="https://github.com/GangLiLab/genekitr/assets/34302878/dd8ce580-cda1-4bd4-b1c4-78084498dfda">


- For `plotGSEA()` function, I have set a parameter 'label_by' that allows users to specify whether to use the 'ID' or 'Description' column as the label for the image.
For example,

```
data(geneList, package = "genekitr")
gs <- geneset::getGO(org = "human",ont = "bp")
gse <- genGSEA(genelist = geneList, geneset = gs)
# default plot
plotGSEA(gse, plot_type = "classic", show_pathway = 'GO:0006260', show_gene = 'SSRP1')
```

<img width="496" alt="image" src="https://github.com/GangLiLab/genekitr/assets/34302878/8e10ef3e-edca-4433-ac99-0cfe69d94bf9">

If user wants to combine GO id and term name, firstly change the "Description" column of gsea_df then specify `label_by='description'`

```
gse$gsea_df$Description = paste0(gse$gsea_df$Hs_BP_ID,' (',gse$gsea_df$Description,')')
plotGSEA(gse, plot_type = "classic", show_pathway = 'GO:0006260', show_gene = 'SSRP1',label_by = 'description')
```

<img width="495" alt="image" src="https://github.com/GangLiLab/genekitr/assets/34302878/31d96003-4aa5-4f49-8e3d-df914e9da60d">

Overall, the new feature added to the function can be obtained in **version >= 1.2.2**
